test-refined/shared/src/main/scala/zio/test/refined/CollectionInstances.scala

```diff
@@ -2,73 +2,108 @@ package zio.test.refined
+
+  def vectorSizeRefinedGen[T: DeriveGen, P](implicit
+    genT: Gen[Random with Sized, T],
+    sizeGen: Gen[Random, Int Refined P]
+  ): Gen[Random with Sized, Refined[Vector[T], Size[P]]] =
+    sizedChunkGen(genT, sizeGen)
+      .map(r => Refined.unsafeApply(r.toVector))
+
   implicit def nonEmptyChunkRefinedDeriveGen[C, T](implicit
     deriveGenT: DeriveGen[T]
   ): DeriveGen[Refined[NonEmptyChunk[T], NonEmpty]] =
     DeriveGen.instance(
-      nonEmptyChunkDeriveGen(deriveGenT).derive.map(Refined.unsafeApply)
+      nonEmptyChunkRefinedGen(deriveGenT.derive)
     )
 
   implicit def nonEmptyListRefinedDeriveGen[T](implicit
     deriveGenT: DeriveGen[T]
-  ): DeriveGen[Refined[List[T], NonEmpty]] =
-    DeriveGen.instance(
-      nonEmptyChunkDeriveGen(deriveGenT).derive
-        .map(v => Refined.unsafeApply(v.toList))
-    )
+  ): DeriveGen[Refined[List[T], NonEmpty]] = DeriveGen.instance(
+    nonEmptyChunkGen(deriveGenT.derive).map(r => Refined.unsafeApply(r.toList))
+  )
 
   implicit def nonEmptyVectorRefinedDeriveGen[T](implicit
     deriveGenT: DeriveGen[T]
   ): DeriveGen[Refined[Vector[T], NonEmpty]] =
     DeriveGen.instance(
-      nonEmptyChunkDeriveGen(deriveGenT).derive
-        .map(v => Refined.unsafeApply(v.toVector))
+      nonEmptyChunkGen(deriveGenT.derive).map(r => Refined.unsafeApply(r.toVector))
     )
 
   implicit def sizedChunkRefinedDeriveGen[T, P](implicit
     deriveGenT: DeriveGen[T],
     deriveGenSize: DeriveGen[Int Refined P]
   ): DeriveGen[Refined[Chunk[T], Size[P]]] =
     DeriveGen.instance(
-      sizedChunkDeriveGen(deriveGenT, deriveGenSize).derive
-        .map(r => Refined.unsafeApply(r))
+      sizedChunkGen(deriveGenT.derive, deriveGenSize.derive).map(r => Refined.unsafeApply(r))
     )
 
   implicit def listSizeRefinedDeriveGen[T, P](implicit
     deriveGenT: DeriveGen[T],
     deriveGenSize: DeriveGen[Int Refined P]
   ): DeriveGen[Refined[List[T], Size[P]]] =
     DeriveGen.instance(
-      sizedChunkDeriveGen(deriveGenT, deriveGenSize).derive
-        .map(r => Refined.unsafeApply(r.toList))
+      sizedChunkGen(deriveGenT.derive, deriveGenSize.derive).map(r => Refined.unsafeApply(r.toList))
     )
 
   implicit def vectorSizeRefinedDeriveGen[T: DeriveGen, P](implicit
     deriveGenT: DeriveGen[T],
     deriveGenSize: DeriveGen[Int Refined P]
   ): DeriveGen[Refined[Vector[T], Size[P]]] =
     DeriveGen.instance(
-      sizedChunkDeriveGen(deriveGenT, deriveGenSize).derive
-        .map(r => Refined.unsafeApply(r.toVector))
+      sizedChunkGen(deriveGenT.derive, deriveGenSize.derive).map(r => Refined.unsafeApply(r.toVector))
     )
 
-  private def nonEmptyChunkDeriveGen[T](implicit
-    arbT: DeriveGen[T]
-  ): DeriveGen[NonEmptyChunk[T]] =
-    DeriveGen.instance(Gen.chunkOf1(arbT.derive))
+  private[refined] def nonEmptyChunkGen[T](
+    arbT: Gen[Random with Sized, T]
+  ): Gen[Random with Sized, NonEmptyChunk[T]] =
+    Gen.chunkOf1(arbT)
```

Comment:
Not sure it Is worth implementing separate helper methods here since you are just called a single method or at most two in the sized variant below.